### DISCUSSION TITLE
OpenVPN configuration files not parsed correctly?

### DISCUSSION THREAD

**Discussion by ObscenityIB** [qdm12/gluetun#2321] [Q&A] [answered]
I am attempting to use the "custom" configuration for mullvad, as the "mullvad" configuration only accepts the username, and not the openvpn config and certificate provided by mullvad.

OpenVPN by itself on my system parses the file fine, but the gluetun docker container doesn't.

One of the first instances of this is when it fails to parse comments:
```
gluetun  | 2024-06-15T13:14:45+10:00 ERROR VPN settings: OpenVPN settings: custom configuration file: extracting information from custom configuration file: extracting connection from file: on line 24: extracting from remote line: remote line has not 2 fields as expected: remote 103.108.229.98 443 # au-mel-ovpn-302
gluetun  | 2024-06-15T13:14:45+10:00 INFO Shutdown successful
```
It is not recognising "#" as a comment marker, and reads the whole line instead, unlike OpenVPN normally does.

This can be bypassed by moving the comment to its own line:
```
# au-mel-ovpn-302
remote 103.108.229.98 443
```

However, the next snag comes up immediately after:

```
gluetun  | 2024-06-15T13:12:09+10:00 ERROR [openvpn] No client-side authentication method is specified.  You must use either --cert/--key, --pkcs12, or --auth-user-pass
gluetun  | 2024-06-15T13:12:09+10:00 INFO [openvpn] Use --help for more information.
```

This error is false, as this is supplied in the configuration file, and OpenVPN normally does not throw this error outside of gluetun.

The config has the line:
```
auth-user-pass /gluetun/mullvad_userpass.txt
```
(yes, the container can find that file)

So there seems to be something wrong with the OpenVPN version, or some config check script gluetun is using, I use OpenVPN 2.6.9 and it works fine.

Anyone have a clue on how to approach this if its not reading the config correctly?

**Comment by qdm12** [comment, marked as answer]
> One of the first instances of this is when it fails to parse comments:

Indeed, fixed in dea4080a7b29c509d7e1690195f96aebb351711e
Next time feel free to create an issue since this is actually a bug 😉 

> This error is false, as this is supplied in the configuration file, and OpenVPN normally does not throw this error outside of gluetun.

This is mentioned in https://github.com/qdm12/gluetun-wiki/blob/main/setup/openvpn-configuration-file.md#added-options
And it's described you should use environment variables for authentication in step 5 of https://github.com/qdm12/gluetun-wiki/blob/main/setup/openvpn-configuration-file.md#setup
